x-pack/plugins/secops/server/lib/events/query.dsl.ts

```diff
@@ -31,20 +31,26 @@ export const eventFieldsMap: Readonly<Record<string, string>> = {
 };
 
 export const buildQuery = (options: EventsRequestOptions) => {
-  const { to, from } = options.timerange;
+  const { limit, cursor, tiebreaker } = options.pagination;
+  const { sortFieldId, direction } = options.sortField;
   const { fields, filterQuery } = options;
-  const esFields = reduceFields(fields, eventFieldsMap);
-  const filter = [
-    ...createQueryFilterClauses(filterQuery as FilterQuery),
-    {
-      range: {
-        [options.sourceConfiguration.fields.timestamp]: {
-          gte: from,
-          lte: to,
+  const esFields = [...reduceFields(fields, eventFieldsMap)];
+  let filter = [...createQueryFilterClauses(filterQuery as FilterQuery)];
```

Comment:
To avoid mutation, would it make sense to change:

`let filter = [...createQueryFilterClauses(filterQuery as FilterQuery)];`

to

`const filterClauses = [...createQueryFilterClauses(filterQuery as FilterQuery)];`

and then on line 42 below https://github.com/elastic/kibana/pull/28368/files#diff-f7b6d59edcdb444dca4247dcde4965f7R42

change:
```
filter = [
      ...filter,
      {
        range: {
          [options.sourceConfiguration.fields.timestamp]: {
            gte: from,
            lte: to,
          },
        },
      },
    ];
```

to:

```
const filter = [
      ...filterClauses,
      {
        range: {
          [options.sourceConfiguration.fields.timestamp]: {
            gte: from,
            lte: to,
          },
        },
      },
    ];
```

?

Reply:
I can see what you want there